Framework/PythonInterface/mantid/api/src/Exports/SpectrumInfo.cpp

```diff
@@ -13,13 +18,53 @@ void export_SpectrumInfo() {
       .def("isMonitor", &SpectrumInfo::isMonitor, (arg("self"), arg("index")),
            "Returns true if the detector(s) associated with the spectrum are "
            "monitors.")
+
       .def("isMasked", &SpectrumInfo::isMasked, (arg("self"), arg("index")),
            "Returns true if the detector(s) associated with the spectrum are "
            "masked.")
+
+      .def("setMasked", &SpectrumInfo::setMasked,
+           (arg("self"), arg("index"), arg("masked")),
+           "Set the mask flag of the spectrum with the given index.")
+
       .def("hasDetectors", &SpectrumInfo::hasDetectors, (arg("self")),
            "Returns true if the spectrum is associated with detectors in the "
            "instrument.")
+
       .def("twoTheta", &SpectrumInfo::twoTheta, (arg("self"), arg("index")),
            "Returns the scattering angle 2 theta in radians w.r.t. beam "
-           "direction.");
+           "direction.")
+
+      .def("signedTwoTheta", &SpectrumInfo::signedTwoTheta,
+           (arg("self"), arg("index")),
+           "Returns the signed scattering angle 2 theta in radians w.r.t. beam "
+           "direction.")
+
+      .def("size", &SpectrumInfo::size, arg("self"),
+           "Returns the number of spectra.")
+
+      .def("hasUniqueDetector", &SpectrumInfo::hasUniqueDetector,
+           (arg("self"), arg("index")),
+           "Returns true if the spectrum is associated with exactly one "
+           "detector.")
+
+      .def("l1", &SpectrumInfo::l1, arg("self"),
+           "Returns the distance from the source to the sample.")
+
+      .def("l2", &SpectrumInfo::l2, (arg("self"), arg("index")),
+           "Returns the distance from the sample to the spectrum.")
+
+      .def("position", &SpectrumInfo::position, (arg("self"), arg("index")),
+           "Returns the position of the spectrum with the given index.")
+
+      .def("sourcePosition", &SpectrumInfo::sourcePosition, arg("self"),
+           "Returns the source position.")
+
+      .def("samplePosition", &SpectrumInfo::samplePosition, arg("self"),
+           "Returns the sample position.")
+
+      .def("spectrumDefinition", &SpectrumInfo::spectrumDefinition,
+           (arg("self"), arg("index")),
+           return_value_policy<copy_const_reference>(),
+           "Returns the SpectrumDefinition of the spectrum.");
```

Comment:
Is it possible to return a Python list of `SpectrumDefinitions`? Perhaps not with how `SpectrumDefinition` works internally, but I think it would be more natural to users.

Reply:
Yeah, that's maybe a better idea than having the user access each definition by index. I'll implement this functionality.

Reply:
@SimonHeybrock I think this was why we had initially implemented the `getAllSpectrumDefinitions` method.

Reply:
@BhuvanBezawada I see, but I still believe it is inconsistent and not a complete solution. I would recommend to not add it as part of this PR. If required, we can work on a larger-scale solution based on an iterator, at a later point in time.

Reply:
Ah I see - I'll take it out of the PR